ui/src/app/components/take-demographic-survey-v2.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+
+import {
+  DEMOGRAPHIC_SURVEY_V2_NOTIFICATION_END_DATE,
+  DEMOGRAPHIC_SURVEY_V2_PATH,
+} from 'app/utils/constants';
+import { profileStore, serverConfigStore, useStore } from 'app/utils/stores';
+
+import { NotificationBanner } from './notification-banner';
+
+export const TakeDemographicSurveyV2BannerMaybe = () => {
+  const { profile } = useStore(profileStore);
+  const [showBanner, setShowBanner] = useState(true);
+
+  const calculateTimeLeft = () => {
+    const difference =
+      +new Date(DEMOGRAPHIC_SURVEY_V2_NOTIFICATION_END_DATE) - +new Date();
+    let timeLeft = {};
+
+    if (showBanner && difference <= 0) {
+      setShowBanner(false);
+    }
+    if (difference > 0) {
+      timeLeft = {
+        days: Math.floor(difference / (1000 * 60 * 60 * 24)),
+        hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
+        minutes: Math.floor((difference / 1000 / 60) % 60),
+        seconds: Math.floor((difference / 1000) % 60),
+      };
+    }
+
+    return timeLeft;
+  };
+
+  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft());
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setTimeLeft(calculateTimeLeft());
+    }, 1000);
+    return () => clearTimeout(timer);
+  });
+
+  let timeLeftDisplayStr = '';
+
+  // Loop through the keys: days, hours, minutes and seconds and append them
+  Object.keys(timeLeft).forEach((interval) => {
+    if (!timeLeft[interval]) {
```

Comment:
or: we could change this from a forEach() to a map(), and move the logic about whether we should display the string elsewhere

Reply:
So this was displaying the text as : 0 days 0hours 1 minute 0 seconds
I added logic so that it can display time as : 1 minute 0 seconds or say 2 hours 0 minute 0 seconds